### ISSUE TITLE
[Bug] Wrong namespace on Model relationship

### ISSUE THREAD

**Issue by promatik** [laravel-shift/blueprint#490]
- Laravel Version: **8.52**
- PHP Version: **7.4.21**
- Blueprint Version: **dev-master**
- Platform: **Windows**

### Issue:

Hi @jasonmccreary!

I'm using blueprint **dev-master** on my project, and I noticed https://github.com/laravel-shift/blueprint/pull/477 (https://github.com/laravel-shift/blueprint/commit/e446a418c28539f5cd0b6404692dd20930523780 commit) introduced a bug.

The bug happens when you **use blueprint config `models_namespace`**, and you have a `.blueprint` generated with `trace` command.

So to reproduce the issue you'll need to;
1) Set `'models_namespace' => 'Models',` on blueprint config.
2) Run the Tracer command to generate the `.blueprint`
3) The following `draft-yaml`;

### draft.yaml:
```yaml
models:
  Post:
    id
    icon_id: foreign:icons.id
    timestamps
```

This `draft.yaml` will generate the following relationship;

```php
public function icon()
{
    return $this->belongsTo(\App\Models\Models\Icon::class, 'icon_id');
}
```

Note the double `Models` on namespace.

---

Although the problem was introduced on that commit, I believe that commit is not the problem. That commit actually fixes a problem with `modelForContext` function.

What I believe to be the problem is the `Trace` command, because it will wrongly (I assume) generate incorrect paths when `models_namespace` config is set.

This is a `.blueprint` file generated with `Trace` command;

```yaml
models:
    Models\Address: { street: 'string nullable', country: string, icon_id: 'integer nullable', monster_id: integer }
    Models\Icon: { name: string, icon: string }
```

when I run the above yaml, the .blueprint is populated with;

```yaml
models:
    Models\Address: { street: 'string nullable', country: string, icon_id: 'integer nullable', monster_id: integer }
    Models\Icon: { name: string, icon: string }
    Post: { icon_id: 'integer unsigned' }
```

The entries generated with `trace` command are relative to `app`, even when `models_namespace` config is set.
The entries generated with `build` are relative to `models_namespace`.

The entries should be relative to `models_namespace`.

Before #477, since that function was not working there was an "hack" on `Tracer` function `relativeClassName` to avoid this issue, by removing that, everything works as expected. I'll submit a PR to aim that.

**Comment by tabacitu** [comment]
TLDR for the above: 

Without PR https://github.com/laravel-shift/blueprint/pull/491, `dev-master` has a bug: 
- IF you use `'models_namespace' => 'Models'` in your config
- AND run `Trace`
- THEN then inside the yaml you will have `Models\Address` as the model, instead of `Address`
 
This creates a disconnect between how things work in `dev-master` as opposed to `v1.24.1`. This bug will then trickle down:
- IF you then generate models, the relationships will be to `\App\Models\Models\Address::class` (wrong)


@ghostwriter - what do you think? never your intention to eliminate the previous behaviour, right?
@jasonmccreary - should @promatik submit a test for https://github.com/laravel-shift/blueprint/pull/491 to be merged? or was this intended for the next version?

PS. This is the only breaking change we found _so far_ in `dev-master` - great work @ghostwriter , we appreciate it, thank you 🙏😀 We'll keep testing `dev-master`, we use its features _daily_ (crazy, I know).

**Comment by promatik** [comment]
Fixed and merged on #491 🙌
